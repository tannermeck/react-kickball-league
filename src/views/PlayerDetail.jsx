

function PlayerDetail(){
    return (
        <h1>Welcome to PlayerDetail</h1>
    )
}
export default PlayerDetail;