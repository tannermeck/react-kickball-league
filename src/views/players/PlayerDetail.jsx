import { useState, useEffect } from 'react';
import { NavLink } from 'react-router-dom';
import { getPlayerById } from '../../services/players';

function PlayerDetail({ match }){
    const { id } = match.params;
    const [player, setPlayer] = useState(null)
    const [loading, setLoading] = useState(true)

    useEffect(() => {
        async function getPlayer(){
            const playerDetail = await getPlayerById(id);
            setPlayer(playerDetail)
            setLoading(false)
        }
        getPlayer()
    }, [id])
    
    if (loading) return <h1>Loading Player...</h1>

    return (
        <>
            <h1>{player.name}</h1>
            <h3>Position: {player.position}</h3>
            <h3>Team: {player.teams.name}</h3>
            <h3>{player.teams.city}, {player.teams.state}</h3>
            <NavLink to={`/players/edit/${player.id}`}>Edit Player</NavLink>
        </>
    )
}
export default PlayerDetail;