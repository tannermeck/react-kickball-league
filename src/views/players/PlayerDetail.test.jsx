import { screen, render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import PlayerDetail from './PlayerDetail';

it('should render out a player detail component', async () => { 
    const { container } = render(
                            <MemoryRouter>
                                <PlayerDetail match={{ params: { id: '3'} }}/>
                            </MemoryRouter>)
    screen.getByText('Loading Player...')              
    const label = await screen.findByText('Position:', {exact: false})
    
    expect(label).toBeInTheDocument();
    expect(container).toMatchSnapshot();

    
})