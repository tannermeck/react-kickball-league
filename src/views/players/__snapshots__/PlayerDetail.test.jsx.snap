// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render out a player detail component 1`] = `
<div>
  <h1>
    Ron Long
  </h1>
  <h3>
    Position: 
    Outfield
  </h3>
  <h3>
    Team: 
    Bridge City Sneakers
  </h3>
  <h3>
    Portland
    , 
    OR
  </h3>
</div>
`;
